cmake/TorchPackage.cmake

```diff
@@ -1,5 +1,26 @@
 # -*- cmake -*-
 
+MACRO(ADD_TORCH_LIBRARY package type src)
+IF ("${type}" STREQUAL "STATIC")
+  if ("${src}" MATCHES "cu$" OR "${src}" MATCHES "cu;")
+    CUDA_ADD_LIBRARY(${package} STATIC ${src})
+  else()
+    ADD_LIBRARY(${package} STATIC ${src})
+  endif()
+ELSE()
+  IF (BUILD_STATIC OR "$ENV{TH_ONLY_STATIC}" STREQUAL "YES")
+    ADD_LIBRARY(${package} ${type} "")
```

Comment:
Ok, it isn't missing `STATIC`, it is missing source files. It was a hacky way to disable dynamic builds when only static versions were required. 

I am going to remove this logic and force push.